File: 2015/04_The_Ideal_Stocking_Stuffer/part_1/main.js
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 4.1
 */

import { createHash } from 'crypto'

let input = "yzbqklnj";

let count = 0;

while(!createHash('md5').update(input + count).digest('hex').startsWith("00000"))
  ++count;

console.log(count);

File: 2015/04_The_Ideal_Stocking_Stuffer/part_1/main.kt
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 4.1
 */
import java.security.MessageDigest

@OptIn(ExperimentalStdlibApi::class)
fun main(args: Array<String>) {
  val input = "yzbqklnj"
  var count = 0

  val md = MessageDigest.getInstance("MD5")

  while(
    !md.digest((input + count).toByteArray())
      .toHexString()
      .startsWith("00000")
  ) ++count

  println(count)
}